crates/linker/src/pe.rs

```diff
@@ -621,4 +751,169 @@ mod test {
+
+        let output = std::process::Command::new(&zig)
+            .current_dir(dir)
+            .args(&[
+                "build-obj",
+                "app.zig",
+                "-target",
+                "x86_64-windows-gnu",
+                "--strip",
+                "-OReleaseFast",
+            ])
+            .output()
+            .unwrap();
+
+        if !output.status.success() {
+            use std::io::Write;
+
+            std::io::stdout().write_all(&output.stdout).unwrap();
+            std::io::stderr().write_all(&output.stderr).unwrap();
+
+            panic!("zig build-obj failed");
+        }
+
+        let mut app = std::fs::read(dir.join("host.exe")).unwrap();
+        let dynamic_relocations = DynamicRelocationsPe::new(&app);
+
+        // Here we manually add the app.obj assembly to the host's .text section
+        // We can "just" do this because the space reserved for this section is bigger
+        // than what is actually used (because PE rounds up sections to a big alignment,
+        // here 0x200)
+        //
+        // So, we append the bytes at the end of the current .text section, redirect the call
+        // to `magic` to the virtual address of the .text section (so right before the new
+        // assembly bytes that we added) and then update the length of the .text section
+
+        let file = PeFile64::parse(app.as_slice()).unwrap();
+        let text_section = file.sections().next().unwrap();
+
+        // end of the code section as an offset into the file
+        let code_section_end =
+            (text_section.file_range().unwrap().0 + text_section.size()) as usize;
+
+        // the virtual address of the end of the text section
+        let code_section_end_virtual = text_section.address() + text_section.size();
+
+        // this is cheating a little bit: these bytes are copied from the `app.obj`
+        // we could get them out programatically of course.
+        let app_bytes: &[u8] = &[0xb8, 0x2a, 0, 0, 0, 0xc3];
+
+        // put our new code into the existing .text section
+        app[code_section_end..][..app_bytes.len()].copy_from_slice(app_bytes);
+
+        redirect_dummy_dll_functions(&mut app, &dynamic_relocations, &[code_section_end_virtual]);
+
+        remove_dummy_dll_import_table(
+            &mut app,
+            dynamic_relocations.data_directories_offset_in_file,
+            dynamic_relocations.imports_offset_in_file,
+            dynamic_relocations.dummy_import_index,
+        );
+
+        // again, a constant that we could get programatically
+        let section_table_offset = 384;
+
+        // update the size of the .text section (which is the first section header)
+        let p = load_struct_inplace_mut::<ImageSectionHeader>(&mut app, section_table_offset);
+        p.virtual_size
+            .set(LE, p.virtual_size.get(LE) + app_bytes.len() as u32);
+
+        std::fs::write(dir.join("hostapp.exe"), app).unwrap()
+    }
+
+    #[ignore]
+    #[test]
```

Comment:
ignored because we don't have `wine` on CI. When we have windows CI, we can add a `cfg(windows)` test here that just runs the `hostapp.exe` binary and  inspects the output.

Reply:
I thought we did have Windows CI, no?
https://github.com/roc-lang/roc/actions/workflows/windows.yml

Reply:
we do now, but not when I put up the PR (it was not yet merged)